checkout/src/test/java/com/checkout/tokenization/repository/TokenRepositoryImplTest.kt

```diff
@@ -423,7 +435,9 @@ internal class TokenRepositoryImplTest {
             val testDispatcher = UnconfinedTestDispatcher(testScheduler)
             Dispatchers.setMain(testDispatcher)
 
-            tokenRepositoryImpl.networkCoroutineScope = CoroutineScope(StandardTestDispatcher(testScheduler))
+            tokenRepositoryImpl.networkCoroutineScope =
+                CoroutineScope(StandardTestDispatcher(testScheduler))
+            tokenRepositoryImpl.dispatcher = testDispatcher
 
```

Comment:
If you follow the suggestion to move the dispatcher to the constructor, we would need to create a new repository to inject the dispatcher:

```
            val testDispatcher = UnconfinedTestDispatcher(testScheduler)
            Dispatchers.setMain(testDispatcher)
            tokenRepositoryImpl = createTokenRepository(testDispatcher)
            tokenRepositoryImpl.networkCoroutineScope = CoroutineScope(StandardTestDispatcher(testScheduler))
```

Where `createTokenRepository` is a private function for to create `TokenRepositoryImpl`

```
private fun createTokenRepository(dispatcher: CoroutineDispatcher = Dispatchers.IO) = TokenRepositoryImpl( // fill the params) 
```